platform/security/sts/security-sts-attributequeryclaimshandler/src/main/java/org/codice/ddf/security/claims/attributequery/AttributeQueryClaimsHandler.java

```diff
@@ -0,0 +1,271 @@
+
+    protected String destination;
+
+    protected static final String ERROR_RETRIEVING_ATTRIBUTES = "Error retrieving attributes from external attribute store [{}] for DN [{}]. ";
+
+    protected static final String ERROR_RETRIEVING_ATTRIBUTES_SEC_LOG = "Error retrieving attributes from external attribute store [%s] for DN [%s]. ";
+
+    public AttributeQueryClaimsHandler() {
+        LOGGER.debug("Starting AttributeQueryClaimsHandler");
+    }
+
+    @Override
+    public List<URI> getSupportedClaimTypes() {
+        LOGGER.debug("Getting supported claim types.");
+        List<URI> supportedClaimTypes = new ArrayList<>();
+        try {
+            for (String claim : supportedClaims) {
+                supportedClaimTypes.add(new URI(claim));
+            }
+        } catch (URISyntaxException e) {
+            LOGGER.error("Error getting supported claim types for external attribute store.", e);
+        }
+        return supportedClaimTypes;
+    }
+
+    @Override
+    public ProcessedClaimCollection retrieveClaimValues(ClaimCollection claims,
+            ClaimsParameters parameters) {
+        Principal principal = parameters.getPrincipal();
+
+        String username = getUsername(principal);
+
+        ProcessedClaimCollection claimCollection = null;
+
+        try {
+            if (!StringUtils.isEmpty(username)) {
+                claimCollection = retrieveExternalClaimValues(username);
+            }
+        } catch (URISyntaxException e) {
+            LOGGER.error(ERROR_RETRIEVING_ATTRIBUTES, externalAttributeStoreUrl, username, e);
+            SecurityLogger.logError(
+                    String.format(ERROR_RETRIEVING_ATTRIBUTES_SEC_LOG, externalAttributeStoreUrl,
+                            username), e);
+        }
+
+        return (claimCollection != null) ? claimCollection : new ProcessedClaimCollection();
+    }
+
+    /**
+     * Retrieves claims from the external attribute store.
+     *
+     * @param username of the user.
+     * @return The collection of claims.
+     * @throws URISyntaxException
+     */
+    protected ProcessedClaimCollection retrieveExternalClaimValues(String username)
+            throws URISyntaxException {
+        ProcessedClaimCollection claimCollection = new ProcessedClaimCollection();
+
+        ProcessedClaimCollection securityClaimCollection = this.getAttributes(username);
+
+        //If security claim collection came back null, return a null claim collection.
+        if (securityClaimCollection == null || securityClaimCollection.size() == 0) {
+            return null;
+        }
+
+        claimCollection.addAll(securityClaimCollection);
+
+        return claimCollection;
+    }
+
+    /**
+     * Parses the CN from the user's DN.
+     *
+     * @param principal of the user.
+     * @return CN of the user.
+     */
+    protected String getUsername(Principal principal) {
+        String cn = null;
+        StringTokenizer st = new StringTokenizer(principal.getName(), ",");
```

Comment:
Yeah, I noticed that. The CN from a certificate is always capital I think, so changing "cn=" to "CN=" should be okay.